src/serac/numerics/equation_solver.cpp

```diff
@@ -502,10 +502,12 @@ class TrustRegion : public mfem::NewtonSolver {
 
     TrustRegionResults  trResults(X.Size());
     TrustRegionSettings settings;
-    settings.min_cg_iterations                          = static_cast<size_t>(nonlinear_options.min_iterations);
-    settings.max_cg_iterations                          = static_cast<size_t>(linear_options.max_iterations);
-    settings.cg_tol                                     = 0.5 * norm_goal;
-    double tr_size                                      = nonlinear_options.trust_region_scaling * std::sqrt(X.Size());
+    settings.min_cg_iterations = static_cast<size_t>(nonlinear_options.min_iterations);
+    settings.max_cg_iterations = static_cast<size_t>(linear_options.max_iterations);
+    settings.cg_tol            = 0.5 * norm_goal;
+
+    scratch        = 1.0;
+    double tr_size = nonlinear_options.trust_region_scaling * std::sqrt(Dot(scratch, scratch));
```

Comment:
If `scratch` is of length `X.size()` then setting each element to 1.0 and taking the inner product with itself is would give the same value as `X.size()` which is still parallel inconsistent?

Reply:
scratch is X.size() on each prof.  Dot() does a parallel reduction over procs to give the total size.  Its kind of an mfem shortcut trick.

Reply:
ahh okay, I was assuming `Dot` was a local operation